src/fhirtypes/ElementDefinition.ts

```diff
@@ -1387,15 +1396,24 @@ export class ElementDefinition {
       if (newElements.length === 0) {
         // If it has a profile, use that, otherwise use the code
         const type = this.type[0].profile?.[0] ?? this.type[0].code;
```

Comment:
I actually assumed it _was_ me.  And I think it is correct.  It's doing what the comment says -- if there is a `profile` defined, prefer that -- otherwise just take the `code`.  We always want to unfold the most specific representation, which in this case would be the profile if it is specified.

We have the `?.[0]` because it's possible that the bit before that is null, so it's null-safe -- and that's the way you do null-safe access to an indexer in TypeScript.